.github/workflows/example-builds-nightly-defaultarch.yml

```diff
@@ -42,3 +42,17 @@ jobs:
           version: ${{ matrix.julia-version }}
       - run: julia --version
       - run: julia --compile=min -O0 -e 'import InteractiveUtils; InteractiveUtils.versioninfo()'
+      - name: "Check that the correct julia is used and that archive mtimes are maintained"
+        run: |
+          if !occursin("hostedtoolcache", Sys.BINDIR)
+            error("the wrong julia is being used: $(Sys.BINDIR)")
+          end
+          if VERSION > v"1.7.0-0" # pkgdir was introduced here, and before then mtime wasn't a problem so just skip
+            using Pkg
+            src = pkgdir(Pkg, "src", "Pkg.jl")
+            # mtime is when it's compressed, ctime is when the file is extracted
+            if !<(mtime(src), ctime(src))
+              error("source mtime ($(mtime(src))) is not earlier than ctime ($(ctime(src)))")
+            end
+          end
+        shell: julia --startup-file=no --color=yes {0}
```

Comment:
You can avoid the repeating this step in multiple workflows by using another job with the [`uses` field](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iduses).

Reply:
Having a common .jl file is effective enough for this I think